File: 1342-number-of-steps-to-reduce-a-number-to-zero/1342-number-of-steps-to-reduce-a-number-to-zero.js
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function(num) {
    // Define step counter
    let steps = 0
    
    // While loop since we have a condition to fulfill and iterations is variable
    while(num > 0){
        // Increment step counter
        steps++
        
        // Test if num is even
        if(num % 2 == 0){
            num /= 2
        } else {
            num -= 1
        }
    }
    return steps
    
};

File: 1342-number-of-steps-to-reduce-a-number-to-zero/1342-number-of-steps-to-reduce-a-number-to-zero.py
class Solution(object):
    def numberOfSteps(self, num):
        """
        :type num: int
        :rtype: int
        """
        steps = 0
        
        while num > 0:
            #Increment step counter for every loop
            steps += 1
            
            # Check if num/2 = true if not subtract 1
            if num % 2 == 0:
                num /= 2
            else:
                num -= 1
        # Return the number of steps needed to reduce        
        return steps
            
        